numpy/distutils/tests/test_system_info.py

```diff
@@ -0,0 +1,190 @@
+from __future__ import division, print_function
+
+import os
+from tempfile import mkdtemp
+
+from numpy.distutils import ccompiler
+from numpy.testing import TestCase, run_module_suite, assert_, assert_equal
+from numpy.distutils.system_info import system_info, ConfigParser
+from numpy.distutils.system_info import default_lib_dirs, default_include_dirs
+
+def get_class(name, notfound_action=1):
+    """
+    notfound_action:
+      0 - do nothing
+      1 - display warning message
+      2 - raise error
+    """
+    cl = {'temp1': test_temp1,
+          'temp2': test_temp2
+          }.get(name.lower(), test_system_info)
+    return cl()
+
+def get_standard_file(fname):
+    """
+    Overrides the get_standard_file from system_info
+    """
+    tmpdir = mkdtemp()
+    filename = os.path.join(tmpdir,fname)
+    with open(filename,'w') as fd:
+        fd.write(site_cfg)
+    filenames = [filename]
+    return filenames
+
+simple_site = """
+[ALL]
+library_dirs = {dir1:s}:{dir2:s}
+libraries = {lib1:s},{lib2:s}
+extra_compile_args = -I/fake/directory
+runtime_library_dirs = {dir1:s}
+
+[temp1]
+library_dirs = {dir1:s}
+libraries = {lib1:s}
+runtime_library_dirs = {dir1:s}
+
+[temp2]
+library_dirs = {dir2:s}
+libraries = {lib2:s}
+extra_link_args = -Wl,-rpath={lib2:s}
+"""
+site_cfg = simple_site
+
+fakelib_c_text = """
+/* This file is generated from numpy/distutils/testing/test_system_info.py */
+#include<stdio.h>
+void foo(void) {
+   printf("Hello foo");
+}
+void bar(void) {
+   printf("Hello bar");
+}
+"""
+
+class test_system_info(system_info):
+    def __init__(self,
+                 default_lib_dirs=default_lib_dirs,
+                 default_include_dirs=default_include_dirs,
+                 verbosity=1,
+                 ):
+        self.__class__.info = {}
+        self.local_prefixes = []
+        defaults = {}
+        defaults['library_dirs'] = []
+        defaults['include_dirs'] = []
+        defaults['runtime_library_dirs'] = []
+        defaults['src_dirs'] = []
+        defaults['search_static_first'] = []
+        defaults['extra_compile_args'] = []
+        defaults['extra_link_args'] = []
+        self.cp = ConfigParser(defaults)
+        self.files = []
+        self.files.extend(get_standard_file('site.cfg'))
+        self.parse_config_files()
+        if self.section is not None:
+            # Have to by-pass a boolean conversion for non-existing variable
+            try:
+                self.search_static_first = self.cp.getboolean(self.section, 'search_static_first')
+            except: pass
+        assert_(isinstance(self.search_static_first, int))
+
+    def _check_libs(self, lib_dirs, libs, opt_libs, exts):
+        """Override _check_libs to return with all dirs """
+        info = {'libraries' : libs , 'library_dirs' : lib_dirs }
+        return info
+
+class test_temp1(test_system_info):
+    section = 'temp1'
+class test_temp2(test_system_info):
```

Comment:
Ok, but what about test_system_info. It makes more sense/easier to follow
if they follow the naming in system_info? Or should I just change
everything to CamelCase?

2015-02-25 9:26 GMT+01:00 Ralf Gommers <notifications@github.com>:

> In numpy/distutils/tests/test_system_info.py
> <https://github.com/numpy/numpy/pull/5597#discussion_r25324231>:
>
> > +        self.parse_config_files()
> > +        if self.section is not None:
> > +            # Have to by-pass a boolean conversion for non-existing variable
> > +            try:
> > +                self.search_static_first = self.cp.getboolean(self.section, 'search_static_first')
> > +            except: pass
> > +        assert_(isinstance(self.search_static_first, int))
> > +
> > +    def _check_libs(self, lib_dirs, libs, opt_libs, exts):
> > +        """Override _check_libs to return with all dirs """
> > +        info = {'libraries' : libs , 'library_dirs' : lib_dirs }
> > +        return info
> > +
> > +class test_temp1(test_system_info):
> > +    section = 'temp1'
> > +class test_temp2(test_system_info):
>
> and another one: class names should be CamelCase
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/numpy/numpy/pull/5597/files#r25324231>.
>



-- 
Kind regards Nick